packages/monorepo/src/components/nx-configurator.ts

```diff
@@ -383,18 +401,56 @@ export class NxConfigurator extends Component implements INxProjectCore {
             "@nx/devkit": "^16",
           },
           private: true,
-          workspaces: this.project.subprojects
+          engines: {
+            node: ">=16",
+            pnpm: ">=8",
+          },
+          scripts: this.project.tasks.all
+            .filter((t) => t.name !== "install")
+            .reduce(
+              (p, c) => ({
+                ...p,
+                [c.name]: NodePackageUtils.command.projen(
+                  NodePackageManager.PNPM,
+                  c.name
+                ),
+              }),
+              {}
+            ),
+        },
+      }).synthesize();
+    }
+
+    if (
+      !ProjectUtils.isNamedInstanceOf(this.project, NodeProject) &&
+      !this.project.tryFindFile("pnpm-workspace.yaml")
+    ) {
+      new YamlFile(this.project, "pnpm-workspace.yaml", {
+        obj: {
+          packages: this.project.subprojects
             .filter((p) => ProjectUtils.isNamedInstanceOf(p, NodeProject))
             .map((p) => path.relative(this.project.outdir, p.outdir)),
         },
       }).synthesize();
     }
   }
 
+  private _invokeInstallCITasks() {
+    const cmd = NodePackageUtils.command.exec(
+      ProjectUtils.isNamedInstanceOf(this.project, NodeProject)
+        ? this.project.package.packageManager
+        : NodePackageManager.NPM,
+      "nx run-many -t install:ci--output-style=stream --nx-bail"
+    );
+    const task = this.project.tasks.tryFind("install:ci");
+    task?.steps[task.steps.length - 1]?.exec !== cmd && task?.exec(cmd);
```

Comment:
I think you'll need to also do `task?.steps?.length` to avoid accessing a property of undefined here :)

Reply:
Fixed.